pkg/blockdevice/v1alpha2/blockdevice.go

```diff
@@ -0,0 +1,178 @@
+*/
+
+package v1alpha2
+
+import (
+	ndmapisv1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/ndm/v1alpha1"
+	apisv1alpha1 "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+)
+
+//TODO: While using these packages UnitTest must be written to corresponding function
+
+// BlockDeviceState is label for block device states
+type BlockDeviceState string
+
+const (
+	// BlockDeviceStateActive is active state of the block device
+	BlockDeviceStateActive BlockDeviceState = "Active"
+)
+
+// DefaultBlockDeviceCount is a map containing the default block device count of various raid types.
+var DefaultBlockDeviceCount = map[string]int{
+	string(apisv1alpha1.PoolTypeMirroredCPV): int(apisv1alpha1.MirroredBlockDeviceCountCPV),
+	string(apisv1alpha1.PoolTypeStripedCPV):  int(apisv1alpha1.StripedBlockDeviceCountCPV),
+	string(apisv1alpha1.PoolTypeRaidzCPV):    int(apisv1alpha1.RaidzBlockDeviceCountCPV),
+	string(apisv1alpha1.PoolTypeRaidz2CPV):   int(apisv1alpha1.Raidz2BlockDeviceCountCPV),
+}
+
+// BlockDevice encapsulates BlockDevice api object.
+type BlockDevice struct {
+	// actual block device object
+	Object *ndmapisv1alpha1.BlockDevice
+}
+
+// BlockDeviceList holds the list of BlockDevice api
+type BlockDeviceList struct {
+	// list of blockdevices
+	ObjectList *ndmapisv1alpha1.BlockDeviceList
+}
+
+// Predicate defines an abstraction to determine conditional checks against the
+// provided block device instance
+type Predicate func(*BlockDevice) bool
+
+// predicateList holds the list of Predicates
+type predicateList []Predicate
+
+// all returns true if all the predicates succeed against the provided block
+// device instance.
+func (l predicateList) all(c *BlockDevice) bool {
+	for _, pred := range l {
+		if !pred(c) {
+			return false
+		}
+	}
+	return true
+}
+
+// HasAnnotation returns true if provided annotation key and value are present
+// in the provided block deive instance.
+func HasAnnotation(key, value string) Predicate {
+	return func(c *BlockDevice) bool {
+		val, ok := c.Object.GetAnnotations()[key]
+		if ok {
+			return val == value
+		}
+		return false
+	}
+}
+
+// TODO: fix below snippet when #1227 merged
+//// IsSparse returns true if the block device is of sparse type.
+//func IsSparse() Predicate {
+//	return func(d *BlockDevice) bool {
+//		return d.Object.GetLabels()[string(apisv1alpha1.NdmBlockDeviceTypeCPK)] == string(apisv1alpha1.TypeBlockDeviceCPV)
+//	}
+//}
+
+// IsType returns true if the block device is of type same as passed argument
+//func IsType(bdType string) Predicate {
+//	return func(d *BlockDevice) bool {
+//		return d.Object.GetLabels()[string(apisv1alpha1.NdmBlockDeviceTypeCPK)] == bdType
+//	}
+//}
+
+// IsActive returns true if the block device is active.
+func IsActive() Predicate {
+	return func(bd *BlockDevice) bool {
+		return bd.Object.Status.State == string(BlockDeviceStateActive)
+	}
+}
+
+// IsUsable returns true if this block device can be used for pool provisioning.
```

Comment:
Have the comments & code within 60 chars or max within 80 chars

Reply:
Done